test/api/conftest.py

```diff
@@ -0,0 +1,104 @@
+from cassandra.policies import RoundRobinPolicy
+
+# By default, tests run against the CQL server listening
+# on localhost:9042. Add the --host and --port options to allow overiding
+# these defaults.  The server listens for API requests on port 10000 by default.
+# To override the API port use the --api-port option.
+def pytest_addoption(parser):
+    parser.addoption('--host', action='store', default='localhost',
+        help='CQL server host to connect to')
+    parser.addoption('--port', action='store', default='9042',
+        help='CQL server port to connect to')
+    parser.addoption('--api-port', action='store', default='10000',
+        help='server API port to connect to')
+    parser.addoption('--ssl', action='store_true',
+        help='Connect to CQL via an encrypted TLSv1.2 connection')
+
+# "cql" fixture: set up client object for communicating with the CQL API.
+# The host/port combination of the server are determined by the --host and
+# --port options, and defaults to localhost and 9042, respectively.
+# We use scope="session" so that all tests will reuse the same client object.
+@pytest.fixture(scope="session")
+def cql(request):
+    profile = ExecutionProfile(
+        load_balancing_policy=RoundRobinPolicy(),
+        consistency_level=ConsistencyLevel.LOCAL_QUORUM,
+        serial_consistency_level=ConsistencyLevel.LOCAL_SERIAL,
+        # The default timeout (in seconds) for execute() commands is 10, which
+        # should have been more than enough, but in some extreme cases with a
+        # very slow debug build running on a very busy machine and a very slow
+        # request (e.g., a DROP KEYSPACE needing to drop multiple tables)
+        # 10 seconds may not be enough, so let's increase it. See issue #7838.
+        request_timeout = 120)
+    if request.config.getoption('ssl'):
+        # Scylla does not support any earlier TLS protocol. If you try,
+        # you will get mysterious EOF errors (see issue #6971) :-(
+        ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLSv1_2)
+    else:
+        ssl_context = None
+    cluster = Cluster(execution_profiles={EXEC_PROFILE_DEFAULT: profile},
+        contact_points=[request.config.getoption('host')],
+        port=request.config.getoption('port'),
+        # TODO: make the protocol version an option, to allow testing with
+        # different versions. If we drop this setting completely, it will
+        # mean pick the latest version supported by the client and the server.
+        protocol_version=4,
+        # Use the default superuser credentials, which work for both Scylla and Cassandra
+        auth_provider=PlainTextAuthProvider(username='cassandra', password='cassandra'),
+        ssl_context=ssl_context,
+    )
+    return cluster.connect()
+
+class ApiSession:
+    def __init__(self, host='locahost', port=10000):
+        self.host = host
+        self.port = port
+        self.session = Session()
```

Comment:
nitpick: I'd use "import requests" and use `requests.Session()` here explicitly. I found it very non-obvious where this "Session" comes from, it's not a very self-explanatory name.

Reply:
can do